helper/resource/testing.go

```diff
@@ -640,6 +642,10 @@ type TestStep struct {
 	// Custom state checks can be created by implementing the [statecheck.StateCheck] interface, or by using a StateCheck implementation from the provided [statecheck] package.
 	ConfigStateChecks []statecheck.StateCheck
 
+	// ConfigQueryResultChecks allow assertions to be made against the query file during a Config test using a query check.
+	// Custom query checks can be created by implementing the [querycheck.QueryResultCheck] interface, or by using a QueryResultCheck implementation from the provided [querycheck] package.
+	ConfigQueryResultChecks []querycheck.QueryResultCheck
```

Comment:
nit: I think for naming, dropping `Config` might be best here. I think it's a little confusing to think of "query" as a part of the `Config` test step mode, since they serve completely different things.

I think it's okay to use the `Config` fields and not explicitly call the query mode part of the "config mode", if that makes sense. Similar to how you can use `Config` with the `ImportState` mode.